test/pmix_test.c

```diff
@@ -141,12 +139,12 @@ int main(int argc, char **argv)
     /* hang around until the client(s) finalize */
     while (!test_terminated()) {
         // To avoid test hang we want to interrupt the loop each 0.1s
-        double test_current;
+        struct timeval time_now, time_elapsed;
 
         // check if we exceed the max time
-        gettimeofday(&tv, NULL);
-        test_current = tv.tv_sec + 1E-6*tv.tv_usec;
-        if( (test_current - test_start) > params.timeout ){
+        gettimeofday(&time_now, NULL);
+        timersub(&time_now, &time_start, &time_elapsed);
```

Comment:
Could you explain what is achieved by this change?
It seems to be exactly the same thing?
If not - can you provide more details about what is wrong with original code.
In my understanding arithmetic should work the same for both big and little endian.

Reply:
@artpol84 It is just a more correct way of subtracting two timeval structs:
```
timersub() subtracts the time value in b from the time value in a, and places the result in the timeval pointed to by res. The result is normalized such that res->tv_usec has a value in the range 0 to 999,999. 
```

Reply:
I wouldn’t say it’s more correct. It’s a different one.
So I tend to keep the code unless it’s wrong (like fixes for the type casting)

Reply:
Code clarity > clever code every single day and twice on Sunday.   The change was also motivated by trying to correct the timing problem I still see on test14.  All other tests are passing now.

Reply:
I don’t see this code to have more clarity. Original way is pretty typical as well.